### DISCUSSION TITLE
Trigger callback on any events

### DISCUSSION THREAD

**Discussion by BinarSkugga** [miguelgrinberg/python-socketio#725] [Q&A]
Small feature request to trigger a callback on any events, like a generic handler. It can currently be achieved by inheriting the AsyncServer class like so:
``` python
import asyncio

from socketio import AsyncServer


class WebSocketServer(AsyncServer):
    async def _trigger_event(self, event, namespace, *args):
        root = self.handlers['/']
        if '*' in root and event not in ['connect', 'disconnect']:
            if asyncio.iscoroutinefunction(root['*']):
                try:
                    ret = await root['*'](event, namespace, *args)
                except asyncio.CancelledError:  # pragma: no cover
                    ret = None
            else:
                ret = root['*'](event, namespace, *args)
            return ret

        return await super()._trigger_event(event, namespace, *args)
```

Then the callback can be added using the following:
```python
@server.on('*')
async def any_event(event, namespace, sid, data):
    pass
```

**Comment by miguelgrinberg** [comment]
You can define a class-based namespace and then override the `trigger_event` method, which is invoked for all received events.

**Comment by BinarSkugga** [comment]
> You can define a class-based namespace and then override the `trigger_event` method, which is invoked for all received events.

For a single namespace. What about all of them ?

**Comment by miguelgrinberg** [comment]
Register the same or different class-based namespace for each of your namespaces.
